packages/engine/tests/experiment/error.rs

```diff
@@ -0,0 +1,150 @@
+use std::{error::Error, fmt, path::PathBuf};
+
+use provider::{Demand, Provider};
+use serde_json::Value;
+
+pub type Result<T, C = TestContext> = error::Result<T, C>;
+
+#[derive(Debug)]
+pub enum TestContext {
+    TestSetup,
+    ExperimentSetup,
+    ExperimentRun,
+    ExperimentOutput,
+}
+
+impl fmt::Display for TestContext {
+    fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::TestSetup => fmt.write_str("Test setup failed"),
+            Self::ExperimentSetup => fmt.write_str("Experiment setup failed"),
+            Self::ExperimentRun => fmt.write_str("Could not finish experiment run"),
+            Self::ExperimentOutput => {
+                fmt.write_str("Experiment output does not match expectations")
+            }
+        }
+    }
+}
+
+impl Provider for TestContext {
+    fn provide<'a>(&'a self, _: &mut Demand<'a>) {
```

Comment:
Are we planning on getting rid of this?

Reply:
Yes:
https://github.com/hashintel/hash/blob/dfd497dfb12b5e18f361f34a3327e624c98fc931/packages/engine/tests/experiment/error.rs#L32